src/transformers/tokenization_utils_base.py

```diff
@@ -1100,35 +1112,73 @@ def additional_special_tokens(self) -> List[str]:
 
     @bos_token.setter
     def bos_token(self, value):
-        self._bos_token = value
+        self._bos_token = (
+            AddedToken(value, normalized=False, rstrip=True, lstrip=True)
+            if isinstance(value, str) and value != ""
+            else value
+        )
```

Comment:
Is the assumption here that: 
* value will be either a value token str, an AddedToken instance or ""?
* Can `AddedToken` have "" as a value? 

Something like the suggestions below would be safer, otherwise `self._bos_token` can be set to anything that isn't an empty string e.g. a tokenizer class

```suggestion
       if isinstance(value, str) and value != "":
           value = AddedToken(value, normalized=False, rstrip=True, lstrip=True)
       elif not isinstance(value, AddedToken): 
           raise ValueError("Cannot set a non-string value as the BOS token")
      
       self._bok_token = value
```